src/sentry/models/authprovider.py

```diff
@@ -48,5 +61,79 @@ def get_provider(self):
 
         return manager.get(self.provider, **self.config)
 
+    def get_scim_token(self):
+        from sentry.models import SentryAppInstallationForProvider
+
+        if self.flags.scim_enabled:
+            return SentryAppInstallationForProvider.get_token(
+                self.organization, f"{self.provider}_scim"
+            )
+        else:
+            return None
+
+    def get_scim_url(self):
+        if self.flags.scim_enabled:
+            return f"https://sentry.io/scim/{self.organization.slug}/scim/v2"
+            # TODO: make dynamic once URL routes are added
+        else:
+            return None
+
+    def enable_scim(self, user):
+        from sentry.mediators.sentry_apps import InternalCreator
+        from sentry.models import SentryAppInstallation, SentryAppInstallationForProvider
+
+        if (
+            not self.get_provider().can_use_scim(self.organization, user)
+            or self.flags.scim_enabled is True
+        ):
+            return
+
+        # check if we have a scim app already
+
+        if SentryAppInstallationForProvider.objects.filter(
+            organization=self.organization, provider="okta_scim"
+        ).exists():
+            logger.info(
+                "scim_installation_exists",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        data = {
+            "name": "SCIM Internal Integration",
+            "author": "Auto-generated by Sentry",
+            "organization": self.organization,
+            "overview": SCIM_INTERNAL_INTEGRATION_OVERVIEW,
+            "user": user,
+            "scopes": [
```

Comment:
I'm wondering if we could tighten up these permissions a bit -- we don't need all of the functionality that `org:admin` has access to for example, but need that scope for editing teams.